modules/cluster/main.tf

```diff
@@ -0,0 +1,160 @@
+# // GKE Cluster
+data "google_compute_network" "default" {
+  name = "default"
+}
+
+data "google_compute_subnetwork" "default" {
+  name   = "default"
+  region = var.gcp_region
+}
+
+data "google_client_config" "default" {}
+
+data "google_artifact_registry_docker_image" "my_image" {
+  location      = var.gcp_region
+  repository_id = var.google_artifact_registry_repository_id
+  image_name    = "agent-image"
+}
+
+# Minimal Autopilot GKE Cluster
+resource "google_container_cluster" "gke" {
+  name             = "ai-agent-cluster"
+  location         = var.gcp_region
+  enable_autopilot = true
+
+  network    = data.google_compute_network.default.id
+  subnetwork = data.google_compute_subnetwork.default.id
+
+  # Set `deletion_protection` to `true` will ensure that one cannot
+  # accidentally delete this instance by use of Terraform.
+  deletion_protection = false
+}
+
+provider "kubernetes" {
+  host = "https://${google_container_cluster.gke.endpoint}"
+
+  token = data.google_client_config.default.access_token
+  cluster_ca_certificate = base64decode(
+    google_container_cluster.gke.master_auth[0].cluster_ca_certificate
+  )
+}
+
+data "sops_file" "hugging_face_secrets" {
+  source_file = "./hugging-face-secret.enc.yaml"
+}
+
+resource "kubernetes_secret_v1" "hugging_face_token" {
+  metadata {
+    name = "hugging-face-token"
+  }
+
+  data = {
+    "HF_TOKEN" = google_secret_manager_secret_version.hugging_face_secret_version.secret_data
+  }
+}
+
+# # https://cloud.google.com/kubernetes-engine/docs/quickstarts/create-cluster-using-terraform#review_the_terraform_files
+# Deploy your container
+resource "kubernetes_deployment_v1" "ai_agent" {
```

Comment:
Can we keep the kubernetes resources out of the module? Should only focus on creating and configuring the cluster (GKE). And we have a separate `kubernetes.tf` within the root directory detailing all the workloads and services to be run within the cluster.

Reply:
soo we should have a separate module for Kubernetes and for the cluster?